build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class VersionRepoUpdater
+    {
+        private static Regex s_nugetFileRegex = new Regex("^(.*?)\\.(([0-9]+\\.)?[0-9]+\\.[0-9]+(-([A-z0-9-]+))?)\\.nupkg$");
+
+        private string _gitHubUser;
+        private string _gitHubEmail;
+        private string _gitHubAuthToken;
+        private string _versionsRepoOwner;
+        private string _versionsRepo;
+
+        public VersionRepoUpdater(
+            string gitHubUser = null,
+            string gitHubEmail = null,
+            string gitHubAuthToken = null,
+            string versionRepoOwner = null,
+            string versionsRepo = null)
+        {
+            _gitHubUser = gitHubUser ?? "dotnet-bot";
+            _gitHubEmail = gitHubEmail ?? "dotnet-bot@microsoft.com";
+            _versionsRepoOwner = versionRepoOwner ?? "dotnet";
+            _versionsRepo = versionsRepo ?? "versions";
+
+            _gitHubAuthToken = gitHubAuthToken ?? Environment.GetEnvironmentVariable("GITHUB_PASSWORD");
```

Comment:
The first question was pertaining to the scope of this class. It sets a bunch of default values to strings, but pulls one default value from an environment variable. I'm wondering if it would make the class more intuitive to use if it required the GitHub auth token to be passed in instead of defaulting to the environment variable.